### ISSUE TITLE
Laptop touchpad scroll area usage lead to the "Bug Detected" dialog

### ISSUE THREAD

**Issue by metacortexplus** [mypaint/mypaint#1117]
#### Description

Laptop touchpad scroll area usage lead to the "Bug Detected" dialog (and this report)

#### Steps to reproduce

HP620 laptop
Fedora 32
use touchpad scroll area (very right part of touchpad)

#### Traceback

```python
Mypaint version: 2.0.0-beta.0+unknown
System information: Linux-5.8.16-200.fc32.x86_64-x86_64-with-glibc2.2.5
Using: Python 3.8.6, GTK 3.24.23, GdkPixbuf 2.40.0, Cairo 1.16.0, GLib 2.64.1
Traceback (most recent call last):
  File "/usr/lib64/mypaint/gui/document.py", line 657, scroll_cb(self=<gui.document.Document object>, tdw=<tileddrawwidget.TiledDrawWidget object at 0x7fcf4f357680 (TiledDrawWidget at 0x55953f7e7130)>, event=<Gdk.EventScroll object at 0x7fcf4d9aff90 (void at 0x5595428f35d0)>)
            dev = event.get_source_device()
            dev_settings = mon.get_device_settings(dev)
            if dev_settings:
  variables: {'dev_settings': (None, []), 'mon.get_device_settings': ('local', <bound method Monitor.get_device_settings of <gui.device.Monitor object at 0x7fcf4e4ed9d0>>), 'dev': ('local', <__gi__.GdkWaylandDevice object at 0x7fcf4e2cbc00 (GdkWaylandDevice at 0x559541e37c50)>)}
  File "/usr/lib64/mypaint/gui/device.py", line 255, get_device_settings(self=<gui.device.Monitor object>, device=<__gi__.GdkWaylandDevice object at 0x7fcf4e2cbc00 (GdkWaylandDevice at 0x559541e37c50)>)
            """
            self._init_device_settings(device)
            return self._device_settings.get(device)
  variables: {'self._init_device_settings': ('local', <bound method Monitor._init_device_settings of <gui.device.Monitor object at 0x7fcf4e4ed9d0>>), 'device': ('local', <__gi__.GdkWaylandDevice object at 0x7fcf4e2cbc00 (GdkWaylandDevice at 0x559541e37c50)>)}
  File "/usr/lib64/mypaint/gui/device.py", line 286, _init_device_settings(self=<gui.device.Monitor object>, device=<__gi__.GdkWaylandDevice object at 0x7fcf4e2cbc00 (GdkWaylandDevice at 0x559541e37c50)>)
                dev_prefs_key = _device_prefs_key(device)
                dev_prefs = self._prefs[_PREFS_ROOT].setdefault(dev_prefs_key, {})
                settings = Settings(dev_prefs)
  variables: {'dev_prefs': (None, []), 'self._prefs': ('local', ObservableDict({'saving.scrap_prefix': '~/MyPaint/scrap', 'input.device_mode': 'screen', 'input.global_pressure_mapping': [(0.0, 1.0), (1.0, 0.0)], 'input.use_barrel_rotation': True, 'input.barrel_rotation_subtract_ascension': True, 'input.barrel_rotation_offset': 0.5, 'view.default_zoom': 1.0, 'view.real_alpha_checks': True, 'ui.hide_menubar_in_fullscreen': True, 'ui.hide_toolbar_in_fullscreen': True, 'ui.hide_subwindows_in_fullscreen': True, 'ui.parts': {'main_toolbar': True, 'menubar': True}, 'ui.feedback.scale': False, 'ui.feedback.last_pos': False, 'ui.toolbar_items': {'toolbar1_file': True, 'toolbar1_scrap': False, 'toolbar1_edit': True, 'toolbar1_blendmodes': False, 'toolbar1_linemodes': True, 'toolbar1_view_modes': True, 'toolbar1_view_manips': False, 'toolbar1_view_resets': True}, 'ui.toolbar_icon_size': 'large', 'ui.dark_theme_variant': True, 'ui.rendered_tile_cache_size': 16384, 'saving.default_format': 'openraster', 'brushmanager.selected_brush': 'deevad/sponge_smudging', 'brushmanager.selected_groups': [], 'frame.color_rgba': (0.12, 0.12, 0.12, 0.92), 'misc.context_restores_color': True, 'document.autosave_backups': True, 'document.autosave_interval': 10, 'display.colorspace_EOTF': 2.2, 'display.colorspace': 'srgb', 'scratchpad.last_opened_scratchpad': '/home/xxxxx/.local/share/mypaint/scratchpads/autosave.ora', 'workspace.layout': {'position': {'x': 50, 'y': 32, 'w': -50, 'h': -100}, 'autohide': True}, 'input.button_mapping': {'Button2': 'PanViewMode', '<Shift>Button2': 'RotateViewMode', '<Primary>Button2': 'ZoomViewMode', '<Alt>Button2': 'ZoomViewMode', '<Primary><Shift>Button2': 'FrameEditMode', '<Shift><Alt>Button2': 'FrameEditMode', 'Button3': 'ShowPopupMenu'}, 'compat_behavior_when_unknown': 'unless-pigment-layer-1.x', 'default_compatibility_mode': '2.x', 'compability_settings': {'1.x': {'pigment_on_by_default': False, 'pigment_layer_is_default': False}, '2.x': {'pigment_on_by_default': True, 'pigment_layer_is_default': True}}, 'colors.history': ['#336699', '#44aa66', '#aa6633', '#003153', '#467a93']})), '_PREFS_ROOT': ('global', 'input.devices'), 'setdefault': (None, []), 'dev_prefs_key': ('local', 'Wayland Finger Scrolling:touchpad:2')}
KeyError: 'input.devices'
```

**Comment by jplloyd** [comment]
Can you reproduce this with v2.0.1?

https://github.com/mypaint/mypaint/releases/tag/v2.0.1

**Comment by metacortexplus** [comment]
> Can you reproduce this with v2.0.1?
> 
> https://github.com/mypaint/mypaint/releases/tag/v2.0.1

No I cant. Works as expected (zoom in/out).

**Comment by jplloyd** [comment]
Glad to hear it.

~~The annoying part is that I have a vague recollection of fixing something in this code, but looking at the diff between 2.0.0 and 2.0.1 (`git diff v2.0.0..v2.0.1 gui lib`) I can't see any changes that look relevant to this. May have been fixed by bumped versions of the dependencies.~~

Never mind, I just realized that you were running the first beta, so my diff check was off by a few months. :smile:
